modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -242,99 +241,147 @@ public SCIMResponse listWithGET(UserManager userManager, String filter, int star
         if (startIndex < 1) {
             startIndex = 1;
         }
-
-        FilterTreeManager filterTreeManager = null;
-        Node rootNode = null;
-        JSONEncoder encoder = null;
         try {
+            // Resolving sorting order.
+            sortOrder = resolveSortOrder(sortOrder, sortBy);
 
-            //check whether provided sortOrder is valid or not
-            if (sortOrder != null) {
-                if (!(sortOrder.equalsIgnoreCase(SCIMConstants.OperationalConstants.ASCENDING)
-                        || sortOrder.equalsIgnoreCase(SCIMConstants.OperationalConstants.DESCENDING))) {
-                    String error = " Invalid sortOrder value is specified";
-                    throw new BadRequestException(error, ResponseCodeConstants.INVALID_VALUE);
-                }
-            }
-            //If a value for "sortBy" is provided and no "sortOrder" is specified, "sortOrder" SHALL default to
-            // ascending.
-            if (sortOrder == null && sortBy != null) {
-                sortOrder = SCIMConstants.OperationalConstants.ASCENDING;
+            // Unless configured returns core-user schema or else returns extended user schema
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+
+            // Build node for filtering.
+            Node rootNode = buildNode(filter, schema);
+
+            // Build node for filtering.
+            JSONEncoder encoder = getEncoder();
+
+            // Get the URIs of required attributes which must be given a value
+            Map<String, Boolean> requiredAttributes = ResourceManagerUtil
+                    .getOnlyRequiredAttributesURIs((SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes,
+                            excludeAttributes);
+
+            // API group should pass a user manager to GroupResourceEndpoint.
+            if (userManager != null) {
+                List<Object> tempList = userManager
+                        .listGroupsWithGET(rootNode, startIndex, count, sortBy, sortOrder, domainName,
+                                requiredAttributes);
+                return processGroupList(tempList, encoder, attributes, excludeAttributes, startIndex);
+            } else {
+                String error = "Provided user manager handler is null.";
+                // Log the error as well.
+                // Throw internal server error.
+                throw new InternalErrorException(error);
             }
+        } catch (CharonException | NotFoundException | InternalErrorException | BadRequestException |
+                NotImplementedException e) {
+            return encodeSCIMException(e);
```

Comment:
is the error being logged in upstream levels? else shall we log here?

Reply:
Fixed in https://github.com/wso2/charon/pull/263